### PR TITLE
[WIP] Log Listener for Automatically pulling Remote Corfu Table Updates to RocksDB

### PR DESCRIPTION
**Pull request by nvaishampayan517**
## Overview

Description:

Why should this be merged: 

Related issue(s) (if applicable): #<number>

## Checklist (Definition of Done):

- [ ] There are no TODOs left in the code
- [ ] [Coding conventions](https://github.com/CorfuDB/CorfuDB/wiki/Corfu-Style-Guidelines) (e.g. for logging, unit tests) have been followed
- [ ] Change is covered by automated tests
- [ ] Public API has Javadoc

### CODE REVIEW

File: infrastructure/src/main/java/org/corfudb/infrastructure/remotecorfutable/loglistener/smr/DeleteOperation.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+/**
+ * This class contains the logic for the delete operation from stream listener to database.
+ *
+ * Created by nvaishampayan517 on 08/19/21
+ */
+@AllArgsConstructor
+public class DeleteOperation implements SMROperation {
+    @NonNull
+    private final ByteString[] args;
```
Comment (xnull) [inline infrastructure/src/main/java/org/corfudb/infrastructure/remotecorfutable/loglistener/smr/DeleteOperation.java:24]: Why not List of ByteStrings?

File: infrastructure/src/main/java/org/corfudb/infrastructure/remotecorfutable/loglistener/smr/DeleteOperation.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    private final UUID streamId;
+
+    /**
+     * {@inheritDoc}
+     * <p>
+     *     Performs delete database operation with given keys.
+     * </p>
+     */
+    @Override
+    public void applySMRMethod(@NonNull DatabaseHandler dbHandler) throws RocksDBException {
```
Comment (xnull) [inline infrastructure/src/main/java/org/corfudb/infrastructure/remotecorfutable/loglistener/smr/DeleteOperation.java:37]: Can we pass DatabaseHandler into the DeleteOperation constructor?

File: infrastructure/src/main/java/org/corfudb/infrastructure/remotecorfutable/loglistener/smr/DeleteOperation.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        if (args.length == 1) {
```
Comment (xnull) [inline infrastructure/src/main/java/org/corfudb/infrastructure/remotecorfutable/loglistener/smr/DeleteOperation.java:38]: Magic number, would be great to know what that means

File: infrastructure/src/main/java/org/corfudb/infrastructure/remotecorfutable/loglistener/smr/DeleteOperation.java [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+            //single delete case
+            RemoteCorfuTableVersionedKey key = new RemoteCorfuTableVersionedKey(args[0], timestamp);
```
Comment (xnull) [inline infrastructure/src/main/java/org/corfudb/infrastructure/remotecorfutable/loglistener/smr/DeleteOperation.java:40]: magic number

File: infrastructure/src/main/java/org/corfudb/infrastructure/remotecorfutable/loglistener/smr/SMROperationFactory.java [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+            ByteBuffer dataBuffer = ByteBuffer.wrap(data.getData());
+            long timestamp = data.getGlobalAddress();
+            LogEntry.LogEntryType dataType = LogEntryPeekUtils.getType(dataBuffer);
+
+            switch (dataType) {
+                case SMR:
+                    return handleSMREntry(streamId, dataBuffer, timestamp);
+                case MULTISMR:
+                    return handleMultiSMREntry(streamId, dataBuffer, timestamp);
+                case NOP: case CHECKPOINT: case MULTIOBJSMR:
```
Comment (xnull) [inline infrastructure/src/main/java/org/corfudb/infrastructure/remotecorfutable/loglistener/smr/SMROperationFactory.java:35]: each `case` on a new line

File: infrastructure/src/main/java/org/corfudb/infrastructure/remotecorfutable/loglistener/smr/UpdateOperation.java [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+    /**
+     * {@inheritDoc}
+     * <p>
+     *     Performs update database operation with given key-value pairs.
+     * </p>
+     */
+    @Override
+    public void applySMRMethod(@NonNull DatabaseHandler dbHandler) throws RocksDBException {
+        int size = args.length;
+        if (size % 2 == 1) {
```
Comment (xnull) [inline infrastructure/src/main/java/org/corfudb/infrastructure/remotecorfutable/loglistener/smr/UpdateOperation.java:39]: magic numbers

File: infrastructure/src/main/java/org/corfudb/infrastructure/remotecorfutable/loglistener/smr/DeleteOperation.java (hunk 2 above)
Comment (nvaishampayan517) [reply infrastructure/src/main/java/org/corfudb/infrastructure/remotecorfutable/loglistener/smr/DeleteOperation.java:37]: What would be the benefit of storing the handler in the Operation object?

File: infrastructure/src/main/java/org/corfudb/infrastructure/remotecorfutable/loglistener/smr/DeleteOperation.java (hunk 1 above)
Comment (nvaishampayan517) [reply infrastructure/src/main/java/org/corfudb/infrastructure/remotecorfutable/loglistener/smr/DeleteOperation.java:24]: The amount of args associated with an operation is fixed while reading from the log entry, so I chose to use a final array.

**Pull request closed without merging**